### ISSUE TITLE
Using the Gameboy Advance's buttons to start ROM and BIOS dumps

### ISSUE THREAD

**Issue by GeigerCounter** [FIX94/gba-link-cable-dumper#17]
Yeah, pretty simple request. I know that the GBA link cable can send button inputs over it that a gamecube game can read since it's a feature of many GBA link cable games. So why not do this instead of requiring a GameCube controller? Amazingly, I don't seem to have one or if I do it's lost and I can't find it, so a solution that doesn't require one would be great and the most elegant and simple solution I can think of is to just let the GBA control things. Since... why not? Also hoping this project isn't dead, it's really awesome.

**Comment by FIX94** [comment]
that would require some logic changes which I dont really feel like doing, I assume you are on a wii and use wiimotes? because I think for some reason I never added that, if you are I guess I can just add support for that instead for controls.

**Comment by GeigerCounter** [comment]
Fair enough, I went and got an after-market controller in the end. I'll leave this issue open in case anyone else feels like putting in a PR. I took a look into it and decided I didn't really feel like doing it either, haha. Yeah though, I'm on wii and I'd appreciate if you did add support for that for future wii users.

**Comment by Magneticdud** [comment]
The gba link cable can't send directly the button inputs, but it's done by a "custom mini game" loaded on the gba memory that sends back the commands back.

So it's not a "pretty simple request" , but a complete rewrite of the gba code that needs to fit the communication part in just 256k ram. Not easy at all

**Comment by DamianS-eng** [comment]
Re: Wiimote input, I've started a fork with changes exclusive to the Wii.dol output file that should allow at least one connectable Wii Remote in the app to navigate alongside a GC controller, with mentions of optionally using their respective buttons on screen. I need to test some more, but for those who need to run this app on a Wii without Gamecube controllers, check it out.

**Comment by TheMasterCoders** [comment]
> The gba link cable can't send directly the button inputs, but it's done by a "custom mini game" loaded on the gba memory that sends back the commands back.
> 
> So it's not a "pretty simple request" , but a complete rewrite of the gba code that needs to fit the communication part in just 256k ram. Not easy at all

lulno
You could easily do this in a couple of lines, see THIS GUY's other project: https://github.com/FIX94/gc-gba-link-cable-demo
Not a complete rewrite buddy :)
